MekHQ/src/mekhq/gui/dialog/AdvanceDaysDialog.java

```diff
@@ -294,7 +302,7 @@ public void startAdvancement(final ActionEvent evt) {
             getGUI().getCampaign().fetchAndClearNewReports();
         }
 
-        setModal(false);
+        setRunning(false);
```

Comment:
Let's put that in a finally block in case we throw in the loop.

Reply:
Better idea: put the internals in a try-catch with logging, then break. It would keep the current functionality and properly handle the remaining days.

Reply:
Works for me.